plugin/src/test/groovy/com/novoda/test/DeployRulesTestRule.groovy

```diff
@@ -0,0 +1,92 @@
+package com.novoda.test
+
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class DeployRulesTestRule implements TestRule {
+
+    List<File> resourceDirs
+    File repoDir
+    String groupId = 'test'
+    String artifactId = 'rules'
+    String version = 'unreleased'
+
+    String getMavenCoordinates() {
+        "$groupId:$artifactId:$version"
+    }
+
+    @Override
+    Statement apply(Statement base, Description description) {
+        new Statement() {
+            @Override
+            void evaluate() throws Throwable {
+                cleanRepo()
+                File projectDir = createProjectDir("${System.currentTimeMillis()}")
+                createBuildScript(projectDir)
+                GradleRunner.create()
+                        .withProjectDir(projectDir)
+                        .withDebug(true)
+                        .forwardStdOutput(new OutputStreamWriter(System.out))
+                        .forwardStdError(new OutputStreamWriter(System.out))
+                        .withArguments('clean', 'publish')
+                        .build()
+                base.evaluate()
+                projectDir.deleteDir()
+            }
+
+        }
+    }
+
+    private void cleanRepo() {
+        def artifactDir = new File(repoDir, "${groupId.replace('.', '/')}/$artifactId/$version")
+        if (artifactDir.exists()) {
+            artifactDir.deleteDir()
+        }
+    }
+
+    private void createBuildScript(File projectDir) {
+        new File(projectDir, 'build.gradle').text = """
+buildscript {
+    repositories {
+        jcenter()
+    }
+}
+
+version='$version'
+
+apply plugin: 'java'
+
+sourceSets {
+    main {
+        resources {
+            srcDirs = ${resourceDirs.collect { "'$it.path'" }}
+        }
+    }
+}
+
+apply plugin: 'maven-publish'
```

Comment:
Note: I went for the [_New Maven Publishing_](https://docs.gradle.org/current/userguide/publishing_maven.html) support. The old one I was used to has a problem with the generation of the metadata/pom file on the first run.